hack/generator/pkg/astmodel/resource.go

```diff
@@ -15,22 +15,23 @@ func CreateResourceDefinitions(name *TypeName, specType *StructType, statusType
 
 	var others []TypeDefiner
 
-	var specName *TypeName
-	if specType != nil {
-		specName = NewTypeName(name.PackageReference, name.Name()+"Spec")
-		specDef, specOthers := specType.CreateDefinitions(specName, idFactory)
+	defineStruct := func(suffix string, structType *StructType) *TypeName {
+		definedName := NewTypeName(name.PackageReference, name.Name()+suffix)
+		defined, definedOthers := structType.CreateDefinitions(definedName, idFactory)
+		others = append(append(others, defined), definedOthers...)
+		return definedName
+	}
 
-		others = append(append(others, specDef), specOthers...)
-	} else {
+	var specName *TypeName
+	if specType == nil {
 		panic("spec must always be provided")
+	} else {
+		specName = defineStruct("Spec", specType)
 	}
```

Comment:
Since the first branch of the **if** contains a panic, you could get rid of the **else** and move the variable declaration down below the test. #extremepedantmode #trivialnit